pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -33,6 +35,10 @@ def __init__(self) -> None:
         self._results = None
         self.global_rank = 0
 
+    @property
```

Comment:
do we think this list of hooks is comprehensive? are there more we want to add down the line?